db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -2750,7 +2753,11 @@ Status DBImpl::WaitForFlushMemTables(
                  (flush_memtable_ids[i] != nullptr &&
                   cfds[i]->imm()->GetEarliestMemTableID() >
                       *flush_memtable_ids[i])) {
-        ++num_finished;
+        if (!wait_for_results_readable ||
```

Comment:
Do you know how WaitForFlushMemTables() could return before a new SV is installed?  I see Flush() code path installs a new MemTableList (https://github.com/facebook/rocksdb/blob/389e66bef56b4f81d3c9683469acb5affc32bd7f/db/flush_job.cc#L331) and install a new SV (https://github.com/facebook/rocksdb/blob/389e66bef56b4f81d3c9683469acb5affc32bd7f/db/db_impl/db_impl_compaction_flush.cc#L302) under the same mutex hold.

Reply:
I'm guessing it's the unlock/relock when writing to Manifest file during LogAndApply: https://github.com/facebook/rocksdb/blob/389e66bef56b4f81d3c9683469acb5affc32bd7f/db/memtable_list.cc#L618

Reply:
I mean when this flush job's version edit finish writing and made the in memory change for the memtables, it could wake up other manifest writers in the manifest write queue: https://github.com/facebook/rocksdb/blob/389e66bef56b4f81d3c9683469acb5affc32bd7f/db/version_set.cc#L5863

And the next manifest writer could start writing, unlock and relock mutex during its manifest file write. At that time, WaitForFlushMemTables() can return while the new SV hasn't been installed yet.

Reply:
This scenario makes sense, thanks.